Standard shader ball asset/scene

crydalch opened the pull request:
![cover](https://github.com/usd-wg/assets/assets/458856/c792abaf-b399-4c52-8672-f97c5399924b)
![figure02](https://github.com/usd-wg/assets/assets/458856/60219887-0f43-4dcd-b52b-01789d8c16c3)

Finally, the standard shader ball PR is here :)

Thanks for everyone's patience and contributions, especially Andre Mazzone and Thomas An. And thanks in advance for input and feedback on the PR.

crydalch commented:
Just a quick note: the `src` directory has the scripts and hip file used to produce all of the images. This is mainly in case higher-res maps/data is needed/desirable, but also just to provide instructions examples. The USD data itself should be considered the "canonical" source, as there will surely be contributions that don't original from Houdini. But if/when that happens, it would be nice (but not a requirement) if those files could also end up in the `src` directory if possible.

jcowles commented:
Apropos the conversation today, does everything pass usdchecker?

AlexSchwank commented on full_assets/StandardShaderBall/layers/environment.usda:
```diff
@@ -0,0 +1,583 @@
+            def Shader "mtlximage1"
+            {
+                uniform token info:id = "ND_image_color3"
+                asset inputs:file = @../../../../../../../tmp@
```
is this still used?

AlexSchwank commented on full_assets/StandardShaderBall/maps/layers/environment.usda:
```diff
@@ -0,0 +1,583 @@
+            def Shader "mtlximage1"
+            {
+                uniform token info:id = "ND_image_color3"
+                asset inputs:file = @../../../../../../../tmp@
```
tmp path here and below again

crydalch replied:
Thanks @AlexSchwank I'm trying to figure out how those are getting authored in there, definitely not meant to be there.

pablode commented:
Great work! 👍 

I'm getting following warning:
```
Warning: in _AddParameterInputsToTerminalNode at line 451 of F:\USD-23.08\pxr\imaging\hdMtlx\hdMtlx.cpp -- NodeDef not found for Node 'standard_surface'
Warning: in HdMtlxCreateMtlxDocumentFromHdMaterialNetworkInterface at line 556 of F:\USD-23.08\pxr\imaging\hdMtlx\hdMtlx.cpp -- Validation warnings for generated MaterialX file.
Node interface error: Input 'emission_color' doesn't match declaration: <standard_surface name="Surface" type="surfaceshader">
```
because in ./layers/shaderball.usda:85, a color3 input is connected to a vector3 image node:
```
color3f inputs:emission_color.connect = </standard_shader_ball_scene/materials/neutral/mtlx/mtlximage2.outputs:out>
<!-- ... -->
def Shader "mtlximage2"
{
  uniform token info:id = "ND_image_vector3"
  float2 inputs:texcoord.connect = </standard_shader_ball_scene/materials/neutral/mtlx/mtlxtexcoord1.outputs:out>
  string inputs:uaddressmode = "clamp"
  string inputs:vaddressmode = "clamp"
  vector3f outputs:out
}
```
(which seems to be a common issue - see https://github.com/AcademySoftwareFoundation/MaterialX/issues/1038)

crydalch commented:
Just pushed up a new update; I squashed the various commits, because I'd let some misplaced files in. But the latest commit has the following updates:

- Fixed some strange tmp paths (thanks @AlexSchwank)
- Fixed type mismatches reported from MaterialX (thanks @pablode)
- Simplified and corrected the camera (thanks @drwave), removed DoF, and re-rendered the images w/ the new camera
- Fixed the broken usdpreview example materials
- Removed the mistaken ‘layers’ folder and usd files from ‘maps’, and the 'material_surface_geo' folder from the layers dir

crydalch commented:
This latest version also passes `usdchecker`, @jcowles  :)

jcowles commented:
Merging! 🎉 🎉 🎉
